lib/ansible/modules/network/iosxr/iosxr_lldp_global.py

```diff
@@ -0,0 +1,371 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2019 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#############################################
+#                WARNING                    #
+#############################################
+#
+# This file is auto generated by the resource
+#   module builder playbook.
+#
+# Do not edit this file manually.
+#
+# Changes to this file will be over written
+#   by the resource module builder.
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+#############################################
+
+"""
+The module file for iosxr_lldp_global
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'network'
+}
+
+DOCUMENTATION = """
+---
+module: iosxr_lldp_global
+version_added: 2.9
+short_description: Manage Global Link Layer Discovery Protocol (LLDP) settings on IOS-XR devices.
+description:
+  - This module manages Global Link Layer Discovery Protocol (LLDP) settings on IOS-XR devices.
+author: Nilashish Chakraborty (@NilashishC)
+options:
+  config:
+    description: The provided global LLDP configuration.
+    type: dict
+    suboptions:
+      holdtime:
+        description:
+          - Specifies the holdtime (in sec) to be sent in packets.
+        type: int
+      reinit:
+        description:
+          - Specifies the delay (in sec) for LLDP initialization on any interface.
+        type: int
+      subinterfaces:
+        description:
+          - Enable or disable LLDP over sub-interfaces.
+        type: bool
+      timer:
+        description:
+          - Specifies the rate at which LLDP packets are sent (in sec).
+        type: int
+      tlv_select:
+        description:
+          - Specifies the LLDP TLVs to enable or disable.
+        type: dict
+        suboptions:
+          management_address:
+            description:
+              - Enable or disable management address TLV.
+            type: bool
+          port_description:
+            description:
+              - Enable or disable port description TLV.
+            type: bool
+          system_capabilities:
+            description:
+              - Enable or disable system capabilities TLV.
+            type: bool
+          system_description:
+            description:
+              - Enable or disable system description TLV.
+            type: bool
+          system_name:
+            description:
+              - Enable or disable system name TLV.
+            type: bool
+  state:
+    description:
+      - The state the configuration should be left in.
+    type: str
+    choices:
+    - merged
+    - replaced
+    - deleted
+    default: merged
```

Comment:
Add the note mentioning the version of iosxr test against and connection type tp use with his module and platform guide reference.
 Refer: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/junos/junos_interfaces.py#L106-L107

Reply:
Applicable for other iosxr resource modules as well

Reply:
@ganeshrn it seems we don't have platform guide reference for IOS-XR